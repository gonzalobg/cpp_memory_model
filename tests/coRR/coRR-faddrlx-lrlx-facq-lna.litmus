C coRR-faddrlx-lrlx-facq-lna
{ [x] = 0; }

P0 (int* x) {
  atomic_fetch_add_explicit(x, 1, memory_order_relaxed);
}

P1 (int* x) {
  int a = atomic_load_explicit(x, memory_order_relaxed);
  atomic_thread_fence(memory_order_acquire);
  if (a == 1) {
    int b = *x;
  }
}

~exists (1:a=1 /\ 1:b=0)