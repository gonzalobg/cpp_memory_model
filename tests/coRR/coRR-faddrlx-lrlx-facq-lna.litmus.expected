Test coRR-faddrlx-lrlx-facq-lna Forbidden
States 2
1:a=0; 1:b=0;
1:a=1; 1:b=1;
Undef
Witnesses
Positive: 2 Negative: 0
Flag *undef*
Condition ~exists (1:a=1 /\ 1:b=0)
Observation coRR-faddrlx-lrlx-facq-lna Never 0 2
Hash=eede93e04d7eeb97be8ae060814154a4

